lldb/source/Core/Address.cpp

```diff
@@ -642,7 +680,11 @@ bool Address::Dump(Stream *s, ExecutionContextScope *exe_scope, DumpStyle style,
                 if (pointer_sc.function != nullptr ||
                     pointer_sc.symbol != nullptr) {
                   s->PutCString(": ");
-                  pointer_sc.DumpStopContext(s, exe_scope, so_addr, true, false,
+                  if(name)
+                    pointer_sc.DumpStopContext(s, exe_scope, so_addr, true, false,
+                                             false, true, true, name);
+                  else
+                    pointer_sc.DumpStopContext(s, exe_scope, so_addr, true, false,
```

Comment:
This may help you see what's going on behind the scenes: https://godbolt.org/z/vxc56798P

You can ignore the assembly specifics but for the fact that fn2 and fn3 generate identical code. Just in the source code one "explicitly" passes nullptr, and one "implicitly" passes it.